lib/dust.js

```diff
@@ -536,8 +543,13 @@
     this.tail = tail;
     this.isObject = head && typeof head === 'object';
     this.head = head;
-    this.index = idx;
-    this.of = len;
+
+    if(idx === undefined && len === undefined) {
+      idx = this.tail && this.tail.index;
+      len = this.tail && this.tail.of;
+    }
+    this.index = this.$idx = idx;
```

Comment:
it's too late now but I wish we exposed a getter instead of the actual value. I doubt it's a big deal but I'm not keen on the "feature" that idx and len are writable. Getting them off of the stack head is a huge improvement so :+1: